Source/Scene/Vector3DTilePrimitive.js

```diff
@@ -279,19 +277,30 @@ define([
         var batchTable = primitive._batchTable;
         var attributeLocations = defaultValue(primitive._attributeLocations, defaultAttributeLocations);
 
+        var pickId = primitive._pickId;
         var vertexShaderSource = primitive._vertexShaderSource;
+        var fragmentShaderSource = primitive._fragmentShaderSource;
         if (defined(vertexShaderSource)) {
             primitive._sp = ShaderProgram.fromCache({
                 context : context,
                 vertexShaderSource : vertexShaderSource,
-                fragmentShaderSource : primitive._fragmentShaderSource,
+                fragmentShaderSource : fragmentShaderSource,
                 attributeLocations : attributeLocations
             });
             primitive._spStencil = primitive._sp;
+
+            fragmentShaderSource = ShaderSource.replaceMain(fragmentShaderSource, 'czm_non_pick_main');
+            fragmentShaderSource =
+                fragmentShaderSource +
+                'void main() \n' +
+                '{ \n' +
+                '    czm_non_pick_main(); \n' +
+                '    gl_FragColor = ' + pickId + '; \n' +
```

Comment:
Why does `VectorTiles` have custom pick programs and commands? Is it not possible to create a derived command from it?

Reply:
Both `VectorTiles` and classification primitives have custom pick commands. The reason is that a single classification primitive will have three draw commands for all of the geometries because they are batched. During picking, each geometry has a different color so it can't be batched (there's three commands for each geometry on pick).

This might change when #6393 is merged.